app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -0,0 +1,881 @@
+                int goal = HPlusCoordinator.getGoal(getDevice().getAddress());
+                byte displayTime = HPlusCoordinator.getScreenTime(getDevice().getAddress());
+                byte country = HPlusCoordinator.getCountry(getDevice().getAddress());
+                byte social = HPlusCoordinator.getSocial(getDevice().getAddress()); // ??
+                byte allDayHeart = HPlusCoordinator.getAllDayHR(getDevice().getAddress());
+                byte wrist = HPlusCoordinator.getUserWrist(getDevice().getAddress());
+                byte alertTimeHour = 0;
+                byte alertTimeMinute = 0;
+
+                if (HPlusCoordinator.getSWAlertTime(getDevice().getAddress())) {
+                    int t = HPlusCoordinator.getAlertTime(getDevice().getAddress());
+
+                    alertTimeHour = (byte) (t / 256);
+                    alertTimeMinute = (byte) (t % 256);
+                }
+
+                byte unit = HPlusCoordinator.getUnit(getDevice().getAddress());
+                byte timemode = HPlusCoordinator.getTimeMode((getDevice().getAddress()));
+
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_COUNTRY,
+                        sex,
+                        age,
+                        bodyHeight,
+                        bodyWeight,
+                        0,
+                        0,
+                        (byte) (goal / 256),
+                        (byte) (goal % 256),
+                        displayTime,
+                        country,
+                        0,
+                        social,
+                        allDayHeart,
+                        wrist,
+                        alertTimeHour,
+                        alertTimeMinute,
+                        unit,
+                        timemode
+                };
+            }
+        });
+        return this;
+    }
+
+    private HPlusSupport setCountry(TransactionBuilder transaction) {
+        LOG.info("Attempting to set country...");
+        transaction.add(new ConditionalWriteAction(ctrlCharacteristic) {
+            @Override
+            protected byte[] checkCondition() {
+
+                byte value = HPlusCoordinator.getCountry(getDevice().getAddress());
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_COUNTRY,
+                        (byte) value
+                };
+            }
+        });
+        return this;
+    }
+
+
+    private HPlusSupport setTimeMode(TransactionBuilder transaction) {
+        LOG.info("Attempting to set Time Mode...");
+        transaction.add(new ConditionalWriteAction(ctrlCharacteristic) {
+            @Override
+            protected byte[] checkCondition() {
+
+                byte value = HPlusCoordinator.getTimeMode(getDevice().getAddress());
+                return new byte[]{
+                        HPlusConstants.COMMAND_SET_PREF_TIMEMODE,
+                        (byte) value
```

Comment:
There's quite a few superfluous casts, probably because the "value" variable was of type int before.